Index: cmake_modules/ConfigurePkgConfig.cmake
===================================================================
--- cmake_modules/ConfigurePkgConfig.cmake
+++ cmake_modules/ConfigurePkgConfig.cmake
@@ -41,7 +41,7 @@
     list(APPEND PKG_CONF_LINK_DEPEND_FILES_TEMP ${libName})
   endforeach()
   if(NOT QNXNTO)
-    list(APPEND PKG_CONF_LINK_DEPEND_FILES_TEMP boost_filesystem-mt boost_signals-mt boost_program_options-mt boost_regex-mt)
+    list(APPEND PKG_CONF_LINK_DEPEND_FILES_TEMP boost_filesystem boost_signals boost_program_options boost_regex)
   else(NOT QNXNTO)
     list(APPEND PKG_CONF_LINK_DEPEND_FILES_TEMP boost_filesystem boost_signals boost_program_options boost_regex)
   endif(NOT QNXNTO)
Index: hrplib/hrpModel/CMakeLists.txt
===================================================================
--- hrplib/hrpModel/CMakeLists.txt
+++ hrplib/hrpModel/CMakeLists.txt
@@ -60,7 +60,7 @@
       hrpCorbaStubSkel-${OPENHRP_LIBRARY_VERSION}
       hrpCollision-${OPENHRP_LIBRARY_VERSION}
       ${OMNIORB_LIBRARIES}
-      boost_filesystem-mt boost_regex-mt
+      boost_filesystem boost_regex
       #${Boost_LIBRARIES}
       dl)
   else(NOT QNXNTO)
Index: hrplib/hrpUtil/CMakeLists.txt
===================================================================
--- hrplib/hrpUtil/CMakeLists.txt
+++ hrplib/hrpUtil/CMakeLists.txt
@@ -65,7 +65,7 @@
   if(NOT QNXNTO)
     target_link_libraries(${target}
       hrpCorbaStubSkel-${OPENHRP_LIBRARY_VERSION}
-      boost_filesystem-mt boost_signals-mt
+      boost_filesystem boost_signals
       #${Boost_LIBRARIES}
       ${PNG_LIBRARY} ${JPEG_LIBRARY} ${ZLIB_LIBRARY}
       ${OMNIORB_LIBRARIES} ${LAPACK_LIBRARIES}
@@ -82,8 +82,8 @@
       )
   endif(NOT QNXNTO)
   if(APPLE)
-  target_link_libraries(${target} boost_system-mt)
-  endif() 
+  target_link_libraries(${target} boost_system)
+  endif()
 elseif(WIN32)
   add_definitions(-DHRP_UTIL_MAKE_DLL)
   set_target_properties(${target} PROPERTIES LINK_FLAGS /NODEFAULTLIB:LIBCMT)
Index: server/ControllerBridge/CMakeLists.txt
===================================================================
--- server/ControllerBridge/CMakeLists.txt
+++ server/ControllerBridge/CMakeLists.txt
@@ -23,8 +23,8 @@
     hrpCorbaStubSkel-${OPENHRP_LIBRARY_VERSION}
     hrpUtil-${OPENHRP_LIBRARY_VERSION}
     ${OPENRTM_LIBRARIES}
-    boost_filesystem-mt boost_program_options-mt boost_regex-mt
-    boost_system-mt
+    boost_filesystem boost_program_options boost_regex
+    boost_system
     )
 elseif(WIN32)
   add_definitions(${OPENRTM_DEFINITIONS})
Index: server/ModelLoader/CMakeLists.txt
===================================================================
--- server/ModelLoader/CMakeLists.txt
+++ server/ModelLoader/CMakeLists.txt
@@ -35,7 +35,7 @@
       ${Boost_SIGNALS_LIBRARY}
       ${Boost_SYSTEM_LIBRARY} )
   else(WIN32)
-    set(extralibraries  ${COLLADA_DOM_LIBRARIES} boost_system-mt boost_thread-mt)
+    set(extralibraries  ${COLLADA_DOM_LIBRARIES} boost_system boost_thread)
   endif(WIN32)
 else( COLLADA_DOM_FOUND )
   set(extralibraries)
