Index: cmake_modules/ConfigurePkgConfig.cmake
===================================================================
--- cmake_modules/ConfigurePkgConfig.cmake	(revision 2628)
+++ cmake_modules/ConfigurePkgConfig.cmake	(working copy)
@@ -42,7 +42,7 @@
   endforeach()
   if(NOT QNXNTO)
     #list(APPEND PKG_CONF_LINK_DEPEND_FILES_TEMP boost_filesystem-mt boost_signals-mt boost_program_options-mt boost_regex-mt)
-    list(APPEND PKG_CONF_LINK_DEPEND_FILES_TEMP ${Boost_FILESYSTEM_LIBRARY} ${Boost_SIGNALS_LIBRARY} ${Boost_PROGRAM_OPTIONS_LIBRARY} ${Boost_REGEX_LIBRARY})
+    list(APPEND PKG_CONF_LINK_DEPEND_FILES_TEMP boost_filesystem boost_signals boost_program_options boost_regex)
   else(NOT QNXNTO)
     list(APPEND PKG_CONF_LINK_DEPEND_FILES_TEMP boost_filesystem boost_signals boost_program_options boost_regex)
   endif(NOT QNXNTO)
Index: hrplib/hrpModel/CMakeLists.txt
===================================================================
--- hrplib/hrpModel/CMakeLists.txt	(revision 2628)
+++ hrplib/hrpModel/CMakeLists.txt	(working copy)
@@ -61,7 +61,7 @@
       hrpCollision-${OPENHRP_LIBRARY_VERSION}
       ${OMNIORB_LIBRARIES}
       #boost_filesystem-mt boost_regex-mt
-      ${Boost_REGEX_LIBRARY}
+      boost_filesystem boost_regex
       #${Boost_LIBRARIES}
       dl)
   else(NOT QNXNTO)
Index: hrplib/hrpUtil/CMakeLists.txt
===================================================================
--- hrplib/hrpUtil/CMakeLists.txt	(revision 2628)
+++ hrplib/hrpUtil/CMakeLists.txt	(working copy)
@@ -66,7 +66,7 @@
     target_link_libraries(${target}
       hrpCorbaStubSkel-${OPENHRP_LIBRARY_VERSION}
       #boost_filesystem-mt boost_signals-mt
-      ${Boost_FILESYSTEM_LIBRARY} ${Boost_SIGNALS_LIBRARY} ${Boost_SYSTEM_LIBRARY}
+      boost_filesystem boost_signals
       #${Boost_LIBRARIES}
       ${PNG_LIBRARY} ${JPEG_LIBRARY} ${ZLIB_LIBRARY}
       ${OMNIORB_LIBRARIES} ${LAPACK_LIBRARIES}
@@ -83,8 +83,8 @@
       )
   endif(NOT QNXNTO)
   if(APPLE)
-  target_link_libraries(${target} boost_system-mt)
-  endif() 
+  target_link_libraries(${target} boost_system)
+  endif()
 elseif(WIN32)
   add_definitions(-DHRP_UTIL_MAKE_DLL)
   set_target_properties(${target} PROPERTIES LINK_FLAGS /NODEFAULTLIB:LIBCMT)
Index: server/ControllerBridge/CMakeLists.txt
===================================================================
--- server/ControllerBridge/CMakeLists.txt	(revision 2628)
+++ server/ControllerBridge/CMakeLists.txt	(working copy)
@@ -23,9 +23,10 @@
     hrpCorbaStubSkel-${OPENHRP_LIBRARY_VERSION}
     hrpUtil-${OPENHRP_LIBRARY_VERSION}
     ${OPENRTM_LIBRARIES}
-    ${Boost_PROGRAM_OPTIONS_LIBRARY} ${Boost_REGEX_LIBRARY}
     #boost_filesystem-mt boost_program_options-mt boost_regex-mt
     #boost_system-mt
+    boost_filesystem boost_program_options boost_regex
+    boost_system
     )
 elseif(WIN32)
   add_definitions(${OPENRTM_DEFINITIONS})
Index: server/ModelLoader/CMakeLists.txt
===================================================================
--- server/ModelLoader/CMakeLists.txt	(revision 2628)
+++ server/ModelLoader/CMakeLists.txt	(working copy)
@@ -35,7 +35,7 @@
       ${Boost_SIGNALS_LIBRARY}
       ${Boost_SYSTEM_LIBRARY} )
   else(WIN32)
-    set(extralibraries  ${COLLADA_DOM_LIBRARIES} boost_system-mt boost_thread-mt)
+    set(extralibraries  ${COLLADA_DOM_LIBRARIES} boost_system boost_thread)
   endif(WIN32)
 else( COLLADA_DOM_FOUND )
   set(extralibraries)
