--- cmake_modules/GetUbuntuVer.cmake    2014-03-21 19:16:03.182840838 -0600
+++ cmake_modules/GetUbuntuVer.cmake    2014-03-21 19:16:22.190814436 -0600
@@ -4,7 +4,7 @@
 SET(UBUNTU_VERSION_NUM "UBUNTU_VERSION_NUM-NOTFOUND")
 SET(UBUNTU_CODENAME "UBUNTU_CODENAME-NOTFOUND")

-if(UNIX AND NOT APPLE AND NOT QNXNTO)
+if(UNIX AND NOT APPLE AND NOT QNXNTO AND EXISTS /etc/lsb-release)
   EXECUTE_PROCESS(COMMAND cat /etc/lsb-release
                   RESULT_VARIABLE  result_val
                   OUTPUT_VARIABLE  output_val
@@ -32,6 +32,6 @@
     STRING(REGEX REPLACE ".*DISTRIB_CODENAME=([^\n]+).*" "\\1"
            UBUNTU_CODENAME ${output_val})
   endif()
-endif(UNIX AND NOT APPLE AND NOT QNXNTO)
+endif(UNIX AND NOT APPLE AND NOT QNXNTO AND EXISTS /etc/lsb-release)
